File: services/Dumble.SharedKernel/Dumble.SharedKernel/Contracts/CurrentUser.cs
using Dumble.SharedKernel.Enums;

namespace Dumble.SharedKernel.Contracts;

public record CurrentUser(
    string Id,
    string Email,
    string DisplayName,
    string? ProfileImage,
    UserType UserType,
    IReadOnlyList<string> Roles
)
{
    public static CurrentUser Empty { get; } =
        new(string.Empty, string.Empty, string.Empty, null, UserType.Participant, Array.Empty<string>());
}

File: services/Dumble.Subscription/src/main/java/com/example/DumbleSubscription/dto/CurrentUser.java
package com.example.DumbleSubscription.dto;

import lombok.Data;

import java.util.List;
import java.util.UUID;

/**
 * The authenticated user attached to {@code SecurityContextHolder.getContext().getAuthentication().getPrincipal()}.
 * Populated by {@link com.example.DumbleSubscription.security.JwtAuthenticationFilter}.
 */
@Data
public class CurrentUser {
    private UUID id;
    private String email;
    private String displayName;
    /** PARTICIPANT | TRAINER | GYM_OWNER | GYM | ADMIN */
    private String userType;
    private List<String> roles;
}
